community/random-values/src/test/java/org/neo4j/values/storable/RandomValuesTest.java

```diff
@@ -0,0 +1,369 @@
+            assertThat( length, lessThanOrEqualTo( 20 ) );
+        }
+    }
+
+    @Test
+    public void nextPrintableAsciiString()
+    {
+        for ( int i = 0; i < ITERATIONS; i++ )
+        {
+            TextValue textValue = randomValues.nextPrintableAsciiString( 10, 20 );
+            String asString = textValue.stringValue();
+            int length = asString.length();
+            assertThat( length, greaterThanOrEqualTo( 10 ) );
+            assertThat( length, lessThanOrEqualTo( 20 ) );
+        }
+    }
+
+    @Test
+    public void nextString()
+    {
+        for ( int i = 0; i < ITERATIONS; i++ )
+        {
+            TextValue textValue = randomValues.nextString( 10, 20 );
+            String asString = textValue.stringValue();
+            int length = asString.codePointCount( 0, asString.length() );
+            assertThat( length, greaterThanOrEqualTo( 10 ) );
+            assertThat( length, lessThanOrEqualTo( 20 ) );
+        }
+    }
+
+    @Test
+    public void nextArray()
+    {
+        HashSet<Class<? extends AnyValue>> seen = new HashSet<>( TYPES );
+        for ( int i = 0; i < ITERATIONS; i++ )
+        {
+            ArrayValue arrayValue = randomValues.nextArray();
+            assertThat( arrayValue.length(), greaterThanOrEqualTo( 1 ) );
+            AnyValue value = arrayValue.value( 0 );
+            assertKnownType( value.getClass(), TYPES );
+            markSeen( value.getClass(), seen );
+        }
+
+        assertThat( seen, empty() );
+    }
+
+    @Test
+    public void nextValue()
+    {
+        HashSet<Class<? extends AnyValue>> all = new HashSet<>( TYPES );
+        all.add( ArrayValue.class );
+        HashSet<Class<? extends AnyValue>> seen = new HashSet<>( all );
+
+        for ( int i = 0; i < ITERATIONS; i++ )
+        {
+            Value value = randomValues.nextValue();
+            assertKnownType( value.getClass(), all );
+            markSeen( value.getClass(), seen );
+        }
+
+        assertThat( seen, empty() );
+    }
+
+    private void assertKnownType( Class<? extends AnyValue> typeToCheck, Set<Class<? extends AnyValue>> types )
+    {
+        for ( Class<? extends AnyValue> type : types )
+        {
+            if ( type.isAssignableFrom( typeToCheck ) )
+            {
+                return;
+            }
+        }
+        fail( typeToCheck + " is not an expected type " );
+    }
+
+    private void markSeen( Class<? extends AnyValue> typeToCheck, Set<Class<? extends AnyValue>> seen )
+    {
+        seen.removeIf( t -> t.isAssignableFrom( typeToCheck ) );
+    }
+
+    private void checkDistribution( Supplier<Value> supplier )
+    {
+        Set<Value> values = new HashSet<>();
+        for ( int i = 0; i < ITERATIONS; i++ )
+        {
+            Value value = supplier.get();
+            assertThat( value, notNullValue() );
+            values.add( value );
+        }
+
+        assertThat( values.size(), greaterThan( 1 ) );
```

Comment:
This is not a very optimistic assertion! :)

Reply:
No it is basically just checking that it does not return the same value each time and that any values returned are non-null. If we are really checking that the distribution is normally distributed it gets more tricky and potentially very flaky. This is more here to make sure we don't do anything stupid and we trust that `java.util.Random` and `java.util.SplittableRandom` gets the distribution of values right.